sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/ScanBuilder.java

```diff
@@ -23,7 +23,7 @@
  * An interface for building the {@link Scan}. Implementations can mixin SupportsPushDownXYZ
  * interfaces to do operator push down, and keep the operator push down result in the returned
  * {@link Scan}. When pushing down operators, the push down order is:
- * filter -&gt; aggregate -&gt; limit -&gt; column pruning.
+ * filter -&gt; aggregate -&gt; sample -&gt; limit -&gt; column pruning.
```

Comment:
shouldn't the sample be the first one? We should follow the natural plan node order of a `SELECT ... FROM ...` statement.